### PR TITLE
feat:create new task

### LINKED ISSUE #200: Create new task
## Description

Implement logic to create new task on the edit page

## Criteria

- [ ] Create a new record on task table

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

Implement a logic to create a new task on the edit page.

## Changes

- Create a new record on `task` table
- Rename `categoryCreationSchema` method to `taskCreationSchema` method
- Call `categoryCreationSchema` method where `taskCreationSchema` method is called

## Review points

1. Set `Content-Type: application/json` on Headers tab in Postman
2. Send a POST method with the following data after selecting raw on Body tab
```
{
  "categoryId": "2cb23f03-da02-4f7f-a275-c32ecf24cf6e",
  "title": "some-title",
  "description": "some-description"
}
```
3. Check whether a new record is added to `task` table

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/app/api/[[...route]]/routes/task.route.ts [hunk 1]
```diff
@@ -9,9 +13,24 @@ app.patch('/:taskId', (c) => {
...
+    const data = c.req.valid('json');
+    console.log('data', data);
+
+    const newTask = await prisma.task.create({
+      data: {
+        title: data.title,
+        description: data.description,
+        categoryId: data.categoryId,
+      },
+    });
```
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/task.route.ts]: Could you add a check for whether the category with the `categoryId` actually exists or not?

File: src/app/api/[[...route]]/routes/task.route.ts [hunk 2]
```diff
@@ -9,9 +13,24 @@ app.patch('/:taskId', (c) => {
   return c.json({ message: `Updating task id: ${taskId}` });
 });
 
-app.post('/:shareHouseId', (c) => {
-  const shareHouseId = c.req.param('shareHouseId');
-  return c.json({ message: `Creating task share house id: ${shareHouseId}` });
+app.post('/', zValidator('json', taskCreationSchema), async (c) => {
+  try {
+    const data = c.req.valid('json');
+    console.log('data', data);
```
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/task.route.ts]: Could you remove the`console.log`?

File: src/app/api/[[...route]]/routes/task.route.ts [hunk 3]
```diff
@@ -9,9 +13,24 @@ app.patch('/:taskId', (c) => {
...
+
+    const newTask = await prisma.task.create({
+      data: {
+        title: data.title,
+        description: data.description,
+        categoryId: data.categoryId,
```
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/task.route.ts]: Could you connect the newly created task to the existing category for relations?

File: src/constants/schema.ts [hunk 4]
```diff
@@ -48,6 +49,19 @@ export const taskCreationSchema = z.object({
...
+ * The schema for the task creation form
+ */
+
+export const taskCreationSchema = z.object({
+  categoryId: z.string(),
+  title: z
+    .string()
+    .min(TASK_TITLE_MIN_LENGTH, minLength('Title', TASK_TITLE_MIN_LENGTH))
+    .max(TASK_TITLE_MAX_LENGTH, maxLength('Title', TASK_TITLE_MAX_LENGTH)),
+  description: z.string(),
```
Comment (akiomatic) [inline src/constants/schema.ts:65]: Could you leave a TODO that says this description needs a correct validation? Also, I'd appreciate it if you could share the necessary changes regarding schemas with the frontend team.

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. Could you check my comments?

File: src/app/api/[[...route]]/routes/task.route.ts (hunk 1 above)
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/task.route.ts]: I added an if statement for that.

File: src/app/api/[[...route]]/routes/task.route.ts (hunk 2 above)
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/task.route.ts]: I removed the `console.log`.

File: src/app/api/[[...route]]/routes/task.route.ts (hunk 3 above)
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/task.route.ts]: I added `connect` to the category.

File: src/constants/schema.ts (hunk 4 above)
Comment (KaiKoide) [reply src/constants/schema.ts:65]: I added todo comment in the method. Also, I am going to share this information with the members.

Comment (akiomatic) [reply src/constants/schema.ts:65]: Where did you add the comment...?

**Review by akiomatic — CHANGES_REQUESTED** [review]
I don't see the comment. Have you pushed the commit?

File: src/constants/schema.ts (hunk 4 above)
Comment (KaiKoide) [reply src/constants/schema.ts:65]: Sorry, I accidentally wrote the comment in the wrong place, so I moved it to the correct location.

**Review by akiomatic — APPROVED** [review]
LGTM.

**Review by akiomatic — APPROVED** [review]

**Pull request merged**